src/controller/python/chip/exceptions/ErrorWrapper.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ErrorWrapper.h"
+
+#include <cstring>
+
+#include <lib/core/CHIPError.h>
+
+using namespace chip;
+
+namespace {
+thread_local ChipError sLastError = CHIP_NO_ERROR;
```

Comment:
Can we please avoid this approach of setting a global variable? It makes assumptions about thread safety which may not always be true in all cases.

I'd rather we return a Python-version of ChipError in every ctypes function, and define new macros to make that conversion easy.